room/runtime/src/main/java/androidx/room/SQLiteCopyOpenHelper.java

```diff
@@ -211,10 +207,66 @@ private void copyDatabaseFile(File destinationFile) throws IOException {
                     + destinationFile.getAbsolutePath());
         }
 
+        // Temporarily open intermediate file database using FrameworkSQLiteOpenHelper and call
+        // open pre-packaged callback. If it fails then intermediate file won't be copied making
+        // invoking pre-packaged callback a transactional operation.
+        callOnOpenPrepackagedDatabaseCallbacks(intermediateFile, writable);
+
         if (!intermediateFile.renameTo(destinationFile)) {
             throw new IOException("Failed to move intermediate file ("
                     + intermediateFile.getAbsolutePath() + ") to destination ("
                     + destinationFile.getAbsolutePath() + ").");
         }
     }
+
+    private Integer readDatabaseVersion(File databaseFile) {
+        try {
+            return DBUtil.readVersion(databaseFile);
+        } catch (IOException e) {
+            Log.w(Room.LOG_TAG, "Unable to read database version.", e);
+            return null;
+        }
+    }
+
+    private void callOnOpenPrepackagedDatabaseCallbacks(File databaseFile, boolean writable) {
+        if(mDatabaseConfiguration.prepackagedCallback != null) {
+
+            SupportSQLiteOpenHelper helper = createFrameworkOpenHelper(databaseFile);
+            try {
+                SupportSQLiteDatabase db = writable ? helper.getWritableDatabase() :
+                        helper.getReadableDatabase();
+
+                mDatabaseConfiguration.prepackagedCallback.onOpenPrepackagedDatabase(db);
+            } catch (SQLiteException e) {
+                throw new RuntimeException("Unable to invoke pre-packaged callback.", e);
+            } finally {
+                // Close the db and let Room re-open it through a normal path
+                helper.close();
+            }
+        }
+    }
+
+    private SupportSQLiteOpenHelper createFrameworkOpenHelper(File databaseFile) {
+        String databaseName = databaseFile.getName();
+        Integer version = readDatabaseVersion(databaseFile);
+        if(version == null) {
```

Comment:
nit: missing space in if: `if (version == null) {`

Reply:
Is there an auto formatting task that I could run or maybe code formatting config file that I could import in AS? The default one seems to be using different rules.

Reply:
I'm not sure, the Studio that is started with `studiow` should have the style configuration. I know we also have some tasks to run `ktlint` but I run them when uploading to Gerrit using the `repo` tool which is different from this Github flow. @dlam might know more about this. It's possible this might not be setup yet, sorry.